app/utils/session.js

```diff
@@ -21,28 +21,43 @@ const validateSessionCookie = function validateSessionCookie (req) {
   if (!chargeId) {
     logger.error('ChargeId was not found in request', getLoggingFields(req))
     return false
-  } else if (!cookie.getSessionCookieName() || !cookie.isSessionPresent(req)) {
+  }
+  const chargeKey = createChargeIdSessionKey(chargeId)
+  if (!cookies.getSessionCookieName() || !cookies.isSessionPresent(req)) {
     logger.info('Session cookie is not present', {
       ...getLoggingFields(req),
       referrer: req.get('Referrer'),
       url: req.originalUrl,
       method: req.method
     })
     return false
-  } else if (!cookie.getSessionVariable(req, 'ch_' + chargeId)) {
+  } else if (!cookies.getSessionVariable(req, chargeKey)) {
     logger.info('ChargeId was not found on the session', {
       ...getLoggingFields(req),
       referrer: req.get('Referrer'),
       url: req.originalUrl,
       method: req.method,
-      session_keys: cookie.getSessionVariableNames(req)
+      session_keys: cookies.getChargesOnSession(req)
```

Comment:
If I understand this right, this is only going to include charges that have a `data` property (i.e. use the new format). Is that going to be confusing?

Reply:
Sure thing.